lib/src/kargs.rs

```diff
@@ -0,0 +1,151 @@
+use anyhow::Ok;
+use anyhow::Result;
+
+use crate::deploy::ImageState;
+use ostree::gio;
+use ostree_ext::ostree;
+use ostree_ext::ostree::Deployment;
+use ostree_ext::prelude::Cast;
+use ostree_ext::prelude::FileEnumeratorExt;
+use ostree_ext::prelude::FileExt;
+
+use serde::Deserialize;
+
+#[derive(Deserialize)]
+struct Config {
+    kargs: Vec<String>,
+    arch: Option<Vec<String>>,
+}
+
+pub(crate) fn get_kargs(
+    repo: &ostree::Repo,
+    booted_deployment: &Deployment,
+    fetched: &ImageState,
+) -> Result<Vec<String>> {
+    let cancellable = gio::Cancellable::NONE;
+    let mut kargs: Vec<String> = vec![];
+
+    // Get the running kargs of the booted system
+    if let Some(bootconfig) = ostree::Deployment::bootconfig(booted_deployment) {
+        if let Some(options) = ostree::BootconfigParser::get(&bootconfig, "options") {
+            let options: Vec<&str> = options.split_whitespace().collect();
```

Comment:
>shared kargs parser

Do you mind elaborating on this? What comes to mind is creating a function called ie. `parse_kargs()` that wraps around `.split_whitepace`, but I'm not sure if that's what you're getting at.

Reply:
Let's use https://docs.rs/ostree/latest/ostree/struct.KernelArgs.html#method.from_string ?

Reply:
I think we had a previous conversation regarding the use of the KernelArgs struct: https://github.com/containers/bootc/pull/401#discussion_r1541557674. The reason we didn't want to use it is because there's no easy way to get all the `key-value` pairs for the kargs.